ego/cli/install.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Edgeless Systems GmbH.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"strings"
+
+	"github.com/klauspost/cpuid/v2"
+	"github.com/zcalusic/sysinfo"
+)
+
+const shellToUse = "bash"
+
+var ErrOSNotSupported = errors.New("os not supported")
+var ErrTargetNotSupported = errors.New("install target not supported")
+var ErrInstallationFail = errors.New("installation fail")
+var ErrExitCodeValue = errors.New("exit code not 0")
+
+// Determine relevant information about os and sgx compatibility. Then pass them with os_details to install
+func (c *Cli) Install(ask func(string) bool, component string) error {
+
+	// sgxlc equivalent to sgxflc ?
+	sgxlc := cpuid.CPU.Supports(cpuid.SGXLC)
+
+	var flcsupport string
+	if sgxlc {
+		flcsupport = "flc"
+	} else {
+		flcsupport = "nonflc"
+	}
+
+	// ERROR-Handling ?
+	var si sysinfo.SysInfo
+	si.GetSysInfo()
+	vendor := si.OS.Vendor
+	version := si.OS.Version
+
+	osDetails := strings.Join([]string{vendor, version, flcsupport}, "-")
+
+	return c.install(ask, os_details, component, "https://raw.githubusercontent.com/edgelesssys/ego/feat/install/src/install.json")
```

Comment:
@Nirusu Do you have an opinion regarding security when using an edglss.cc URL here?

Reply:
Mhh, it would require us to follow redirects and trust another operator (TinyURL).

It's not really an user-visible URL though... The only benefit I can see is that if we ever decide to change the path of our repo / the JSON file, it can stay compatible (...maybe?). Other than that, better to keep it pointed to GitHub directly.